mcs_benchmark_data/pipelines/commonsense_qa/roberta_commonsense_qa_submission_pipeline.py

```diff
@@ -10,19 +12,26 @@
 )
 
 
-class RobertaCommonsenseQaSubmissionPipeline(_Pipeline):
+class RobertaCommonsenseQaSubmissionPipeline(_SubmissionPipeline):
     BENCHMARK_ID = "commonsense_qa"
     SUBMISSION_ID = "roberta"
 
-    def __init__(self, data_dir_path: Path = DATA_DIR_PATH, **kwds):
-        _Pipeline.__init__(
+    def __init__(
+        self,
+        *,
+        data_dir_path: Path = DATA_DIR_PATH,
+        dataset_type: str = _SubmissionPipeline.DATASET_TYPE_DEFAULT,

```

Comment:
If the default is the same for all pipelines (and it is, here) and you're just passing it on to one place (the parent class, here), then you don't need to declare this parameter here at all.

```
class SomeSubmissionPipeline(_SubmissionPipeline):
    def __init__(self, ..., **kwds): _SubmissionPipeline.__init__(self, ..., **kwds)
class _SubmissionPipeline(_Pipeline):
    def __init__(self, ..., dataset_type=DATASET_TYPE_DEFAULT, **kwds): _Pipeline.__init__(self, ..., **kwds)
```
Then you're only declaring the default in one place.

You couldn't do this with `data_dir_path` because you have to pass it to both the transformer and the pipeline. That means either both parent classes need the default declared or this class needs the default declared.

Hope that makes sense.

Reply:
So would I then delete the transformer declaration and place it in the _SubmissionPipeline init? I am a bit confused as to how I could pass the specialized transformer (e.g. KagnetCommonsenseQaSubmissionTransformer) to the _SubmissionPipeline without having a dataset_type to give it.

Reply:
My explanation was confusing. I think it'd be easiest just to leave this code as it is for now.

Reply:
Ok sounds good.